TRUtil/LaterClassicAutosplitter.cs

```diff
@@ -44,6 +45,30 @@ public override bool ShouldReset(LiveSplitState state)
         return loadingIntoMainMenu && comingFromALevel;
     }
 
+    protected bool DeathrunShouldSplit()
+    {
+        bool laraJustDied = Data.Health.Old > 0 && Data.Health.Current <= 0;
+        return laraJustDied;
+    }
+
+    protected bool SecretShouldSplit()
+    {
+        if (!Data.Secrets.Changed)
+            return false;
+
+        uint currentGfLevelComplete = Data.GfLevelComplete.Current;
+        uint oldGfLevelComplete = Data.GfLevelComplete.Old;
+
+        if (currentGfLevelComplete != oldGfLevelComplete)
+            return false;
+
+        if (currentGfLevelComplete != 0)
+            return false;
+
+        bool secretWasTriggered = Data.Secrets.Current > Data.Secrets.Old;
+        return secretWasTriggered;
```

Comment:
Isn't this gonna trigger when people load old save files to do warps?

Scenario:
Make a save in level A with 5 secrets.
Go to level B with 10 secrets. Embed and save to prepare a warp.
Load the save from level A.
Load the save from level B => warp triggers, but Secrets.Current > Secrets.Old.

Maybe this is why you check for level complete above.

Reply:
You were right on both accounts! The intention of the guards was to prevent this, however it failed. In fact, the loaded save's `Secrets` value is adopted before the loading screen fades in and even before some other values such as Lara's health, etc. begin to zero/re-initialize.